### PR TITLE
ANDROAPP-5829-adapt-input-image-to-new-mobile-ui-functionality

### PR DESCRIPTION
**Pull request by msasikanth**

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/listView/SearchTEList.kt [hunk 1]
```diff
@@ -232,8 +231,13 @@ class SearchTEList : FragmentGlobalAbstract() {
 
     // TODO replace ImageDetailBottomDialog with new mobile ui Image detail component
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/listView/SearchTEList.kt:232]: please remove this line

File: form/src/main/java/org/dhis2/form/ui/FormView.kt [hunk 2]
```diff
@@ -963,8 +974,13 @@ class FormView : Fragment() {
 
     // TODO replace ImageDetailBottomDialog with new mobile ui Image detail component
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/ui/FormView.kt:975]: Please, remove this commented line

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/mapView/SearchTEMap.kt [hunk 3]
```diff
@@ -260,14 +259,15 @@ class SearchTEMap : FragmentGlobalAbstract(), MapboxMap.OnMapClickListener {
...
                 if (binding.mapCarousel.carouselEnabled) {
-                    // TODO replace ImageDetailBottomDialog with new mobile ui Image detail component
-                    ImageDetailBottomDialog(
-                        null,
-                        File(path),
-                    ).show(
-                        childFragmentManager,
-                        ImageDetailBottomDialog.TAG,
-                    )
+                    if (!path.isNullOrBlank()) {
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/mapView/SearchTEMap.kt:262]: this nested if is generating a code smell, maybe you can merge them

File: commons/src/main/java/org/dhis2/commons/dialogs/imagedetail/ImageDetailActivity.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                title = title.orEmpty(),
+                onDismiss = { finish() },
+                onDownloadButtonClick = { saveImage(imagePath) },
+                onShareButtonClick = { shareImage(imagePath) },
+            )
+        }
+    }
+
+    private fun saveImage(image: String) {
+        // TODO: Implement saving image to disk
```
Comment (andresmr) [inline commons/src/main/java/org/dhis2/commons/dialogs/imagedetail/ImageDetailActivity.kt:56]: I would develop a new download image implementation given that the one in Qr it seem is not working properly, and then this new implementation could be reused in other components like QR one. What do you think?

**Review by andresmr — CHANGES_REQUESTED** [review]

File: commons/src/main/java/org/dhis2/commons/dialogs/imagedetail/ImageDetailActivity.kt (hunk 4 above)
Comment (msasikanth) [reply commons/src/main/java/org/dhis2/commons/dialogs/imagedetail/ImageDetailActivity.kt:56]: Yeah, we would have to save the image publicly accessible folder like downloads. That should make the implementation simpler. We can do this in another story?

We also need to replace it in other places where the `InputImage` component is used as well (in Forms?), so we can do replace all the save image functionality in a single story. WDYT?

Comment (andresmr) [reply commons/src/main/java/org/dhis2/commons/dialogs/imagedetail/ImageDetailActivity.kt:56]: yes, it is ok. so we could merge this without implementation and the develop the other one separately

**Review by andresmr — APPROVED** [review]

**Review by xavimolloy — APPROVED** [review]

**Pull request merged**